apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/queries.ex

```diff
@@ -1060,7 +1060,7 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Queries do
   def retrieve_realms! do
     statement = """
     SELECT *
-    FROM astarte.realms
+    FROM #{Config.astarte_instance_id!()}astarte.realms
```

Comment:
This should be computed using ` CQLUtils.realm_name_to_keyspace_name("astarte", Config.astarte_instance_id!())` rather than just interpolating the instance id

Reply:
with 
` CQLUtils.realm_name_to_keyspace_name("astarte", Config.astarte_instance_id!())` 
it becames "astarteastarte.realms" breaking compatibility with old deployments

with  
`#{Config.astarte_instance_id!()}astarte.realms`
assuming an astarte_instance_id set as "" (old deployments) it becames "astarte.realms", preserving compatibility

Reply:
The interpolation `#{Config.astarte_instance_id!()}astarte.realms` leaks to the external world the implementation detail that an Astarte instance id must be shorter than 41 characters because of Scylla keyspace name length limit. This is not a business logic detail, and therefore it should be internally handled by Astarte.

As you also say, `#{CQLUtils.realm_name_to_keyspace_name("astarte", Config.astarte_instance_id!())}.realms` is backward-compatible if the instance id is set to `""`, as it will be for old Astartes.

I do not see the added value of keeping a different logic just for the `astarte` keyspace, since it both increases the complexity of the solution (by having two ways for keyspace names to be generated)  and adds an implicit constraint on the deployment (the instance id length)